crates/nu-parser/src/parse_command.rs

```diff
@@ -115,7 +115,10 @@ pub fn parse_command_tail(
             positional = positionals;
         }
         Err(reason) => {
-            if found_error.is_none() && !tail.source().contains("help") {
+            if found_error.is_none()
+                && !tail.source().contains("help")
+                && !tail.source().contains("h")
```

Comment:
Would this change for anything that has an 'h' in it?

Reply:
I haven't gone through the entire parser logic tbh. Can you help me figure out a case where you think this might break?

Reply:
`echo "a,b,c" | from-csv --headerless asjdasd` would be a good place to start. Right now this happens:

```
error: from-csv unexpected asjdasd
- shell:1:44
1 | echo "a,b,c" | from-csv --headerless asjdasd
  |                                      ^^^^^^^ unexpected argument (try from-csv -h)
```

`--headerless` contains a `h`, though, so it could potentially be impacted by your change.

Reply:
I see what you mean. I've updated the patch to check for named arguments instead of just string match. I think that would be a better check. Let me know what you think.